packages/modal/addon/utils/event-handler.js

```diff
@@ -0,0 +1,72 @@
+import { isPresent } from '@ember/utils';
+
+export default {
+  /**
+   * Binds the given callback with context to 'eventName' on the element / document and returns the reference.
+   * 
+   ```js
+    import Component from '@ember/component';
+    import EventHandler from '@freshworks/modal/utils/event-handler';
+    export default Component.extend({
+      state: null,
+      action: {
+        setFocusListener() {
+          let _focusListener = EventHandler.bindEvent({
+            context: this, 
+            eventName: 'focusin',
+            callback: this.loopFocus
+          });
+          set(this, '_focusListener', _focusListener);
+        }
+      }
+    });
+  ```
+   *
+   * @function bindEvent
+   * @public
+   * @param {string} eventName Name of the event to bind the callback to
+   * @param {function} callback Callback function for the eventhandler 
+   * @param {string} element element selector. Defaults to document
+  */
+  bindEvent({ eventName, callback, element } = {}) {
+    let domElement = isPresent(element) ? document.querySelector(element) : document;
+    
+    if(isPresent(domElement)) {
+      domElement.addEventListener(eventName, callback);
+    }
+  },
```

Comment:
return callback so that it can be consumed by the caller. Technically, something similar to `.addEventListener`